greedy_max/schedule.py

```diff
@@ -1,65 +1,125 @@
                  name: str,
                  band: Band,
-                 site: Site,
+                 category: Category,
+                 observed: int, # observation time on time slots
+                 length: int, # acq+time (this need to be change)
                  instrument: str,
                  disperser: str,
-                 priority: str,
-                 time: Quantity,
-                 total_time: Quantity,
-                 completion: Quantity):
+                 acquisition: int
+                 ) -> None:
+        self.idx = idx
         self.name = name
-        self.idx = Observation._counter
-        self.site = site
         self.band = band
+        self.category = category
+        self.observed = observed
+        self.length = length
         self.instrument = instrument
         self.disperser = disperser 
-        self.priority = priority
-        self.time = time
-        self.total_time = total_time
-        self.completion = completion  
+        self.acquisition = acquisition
+    
+    def __str__(self) -> str:
+        return f'{self.idx}-{self.name}'
 
-        Observation._counter += 1
+class SchedulingUnit:
+    def __init__(self,
+                 idx: int,
+                 site: Site, 
+                 observations: List[Observation], 
+                 calibrations: List[Observation],
+                 can_be_split: bool,
+                 standard_time: int
+                 ) -> None:
+        self.idx = idx
+        self.site = site
+        self.observations = observations # group or a single science observation
+        self.calibrations = calibrations # group or a single cal observation
+        self.can_be_split = can_be_split # split flag
+        self.standard_time = standard_time # standard time in time slots 
+        #self.priority = priority # ToO or not? 
+        #self.length = self._length()
+        #self.observed = self._observed()
 
-    def calibrate(self, dt):
-        if self.disperser and self.instrument:
-            tcal = 18.0 * u.min if self.disperser != 'mirror' and \
-                            self.disperser != 'null' and \
-                            self.instrument != 'GMOS' else 0.0 * u.min
-            # save to total time or add units
-            return int(np.ceil(tcal.to(u.h) / dt.to(u.h))) 
-        return 0
+    def length(self) -> int:
+        """
+        Calculate the length of the unit based on both observation and calibrations times
+        """
+        obs_slots = sum([obs.length for obs in self.observations])
 
+        if self.standard_time > 0: # not a single science observation
+            standards_needed = max(1, int(obs_slots // self.standard_time))
 
-# TODO: Implement scheduling groups
-# class SchedulingGroup:
-#    def __init__():
+            if standards_needed == 1:
+                cal_slots = self.calibrations[0].length #take just one
+            else:
+                cal_slots = sum([cal.length for cal in self.calibrations])
```

Comment:
if `standards_needed` is 1, I only need to consider the first calibration, while in any other case I have to take in account n calibrations. So if do `cal_slots = sum(calibration.length for calibration in self.calibrations)` while `standards_needed = 1` and `len(calibrations) > 0` then I'm considering more than one calibration on the length and this break GreedyMax. I talk with Bryan on Slack(June 22nd) actually because the first version I implemented was as you suggest but the length of the observation was off.